### PR TITLE
Added field name capitalization to discover and visualize

### PR DESCRIPTION
**Pull request by alexweltman**
* Added field name capitalization to queries on the discover and visualize tabs
* fixed broken time ranges being sent to audit log

### CODE REVIEW

File: src/kibana/plugins/dashboard/index.js [hunk 1]
```diff
@@ -70,182 +71,171 @@ define(function (require) {
...
+          controller: function ($scope, $route, $routeParams, $location, $http, 
+             configFile, Private, getAppState, saveRuleManager, indexPatterns) {
+            indexPatterns.refresh();
+            $scope.saveRuleManager = saveRuleManager;
+            var queryFilter = Private(require('components/filter_bar/query_filter'));
 
-        var notify = new Notifier({
-          location: 'Dashboard'
-        });
+            var notify = new Notifier({
```
Comment (alexweltman) [inline src/kibana/plugins/dashboard/index.js]: here I just removed the link function cause it's no longer needed, that functionality was moved to the searchAuditor service and to $scope.filterResults.  The rest of the changes is messed up formatting from my editor which I will push up a fix for

File: src/kibana/netmon_libs/custom_modules/save_rule/services/searchAuditor.js [hunk 2]
```diff
@@ -1,12 +1,20 @@
...
-    app.factory('searchAuditor', function (elasticSearchFields,
-                                           Restangular) {
-        var previousQuery = {
-            query    : "",
-            fromDate : "",
-            toDate   : ""
-        };
+    var _ = require('lodash');
+    var moment = require('moment');
+    var angular = require('angular');
```
Comment (sammccord) [inline src/kibana/netmon_libs/custom_modules/save_rule/services/searchAuditor.js]: Why are we requiring angular?

Comment (alexweltman) [inline src/kibana/netmon_libs/custom_modules/save_rule/services/searchAuditor.js]: kibana does this on every file that uses an angular method like angular.copy

**Comment by sammccord** [comment]
:shipit:

**Comment by KjellKod** [comment]
ha ha ha ... Is that a plus one @sammccord ?

**Comment by KjellKod** [comment]
@alexweltman is this for 3.1.2?

**Comment by KjellKod** [comment]
👍  reviewed & tested

**Pull request merged**